src/app/components/SearchResults/ResultsRow.jsx

```diff
@@ -20,32 +21,57 @@ const ResultsRow = (props) => {
     return null;
   }
 
-  const items = props.rows.map(item => (
-    {
+  const items = props.rows.map((item) => {
+    let publisher = item && item.agents && item.agents.find(i => i.roles.indexOf('publisher') > -1);
+    publisher = publisher && publisher.name;
+    return {
       ebooks: (item.items) ? item.items : [],
-      pub_date: (item.pub_date) ? parseInt(item.pub_date) : null,
+      pub_date: (item.pub_date_display) ? item.pub_date_display : null,
       pub_place: (item.pub_place) ? item.pub_place : null,
-      publisher: (item.publisher) ? item.publisher : null,
-    }
-  ));
+      publisher,
+    };
+  });
 
   return (
     <ul className="nypl-items-list">
       {
-        items.map((element, key) => (
-          <li className="nypl-results-item" key={key.toString()}>
-            <div className="nypl-results-description">
-              {(element.ebooks) ? <EBookList ebooks={element.ebooks} eReaderUrl={props.eReaderUrl} /> : ''}
-              <span className="nypl-results-date">{element.pub_date}</span>
-              <span className="nypl-results-place">{element.pub_place}</span>
-              <span className="nypl-results-publisher">{element.publisher}</span>
-              <span className="nypl-results-info">{element.language}</span>
-            </div>
-          </li>
-        ))
+        items.map((element, key) => {
```

Comment:
Looks like I didn't clean the `element` name up here. Should use `item` instead for all refs to `element`.

Reply:
ready